xdist/remote.py

```diff
@@ -111,18 +111,18 @@ def pytest_logwarning(self, message, code, nodeid, fslocation):
 
 def serialize_report(rep):
     def disassembled_report(rep):
-        reprtraceback = rep.longrepr.reprtraceback.__dict__
-        reprcrash = rep.longrepr.reprcrash.__dict__
+        reprtraceback = rep.longrepr.reprtraceback.__dict__.copy()
+        reprcrash = rep.longrepr.reprcrash.__dict__.copy()
 
         new_entries = []
         for entry in reprtraceback['reprentries']:
             entry_data = {
                 'type': type(entry).__name__,
-                'data': entry.__dict__,
+                'data': entry.__dict__.copy(),
             }
             for key, value in entry_data['data'].items():
                 if hasattr(value, '__dict__'):
-                    entry_data['data'][key] = value.__dict__
+                    entry_data['data'][key] = value.__dict__.copy()
```

Comment:
`reprcrash` and `value.__dict__` actually don't need to be copied but I felt including it might safe us from inadvertent mutations down the line. I can pull it of course.

Reply:
I think that's fine.